admob/src_java/com/google/firebase/admob/internal/cpp/InterstitialAdHelper.java

```diff
@@ -91,15 +92,18 @@ public void initialize(final long callbackDataPtr, Activity activity, String adU
           public void run() {
             int errorCode;
             String errorMessage;
+            errorCode = ConstantsHelper.CALLBACK_ERROR_ALREADY_INITIALIZED;
+            errorMessage = ConstantsHelper.CALLBACK_ERROR_MESSAGE_ALREADY_INITIALIZED;
             if (mInterstitial == null) {
-              errorCode = ConstantsHelper.CALLBACK_ERROR_NONE;
-              errorMessage = ConstantsHelper.CALLBACK_ERROR_MESSAGE_NONE;
-              mInterstitial = new InterstitialAd(mActivity);
-              mInterstitial.setAdUnitId(mAdUnitId);
-              mInterstitial.setAdListener(new InterstitialAdListener());
-            } else {
-              errorCode = ConstantsHelper.CALLBACK_ERROR_ALREADY_INITIALIZED;
-              errorMessage = ConstantsHelper.CALLBACK_ERROR_MESSAGE_ALREADY_INITIALIZED;
+              try {
+                mInterstitial = new InterstitialAd(mActivity);
+                mInterstitial.setAdUnitId(mAdUnitId);
+                mInterstitial.setAdListener(new InterstitialAdListener());
+                errorCode = ConstantsHelper.CALLBACK_ERROR_NONE;
+                errorMessage = ConstantsHelper.CALLBACK_ERROR_MESSAGE_NONE;
+              } catch(IllegalStateException ise) {
```

Comment:
True, it has something to do with using a wrong or changed AdUnitId. But by leaving a possible exception uncaught it ends up crashing JNI instead of failing the future